### ISSUE TITLE
Add function getEntryByKey()

### ISSUE THREAD

**Issue by gorets** [raulferras/PHP-po-parser#70]
Pleace add new method in Sepia\PoParser\Catalog

```
public function getEntryByKey($key)
    {
        if (!isset($this->entries[$key])) {
            return null;
        }

        return $this->entries[$key];
    }
```

**Comment by raulferras** [comment]
Just wondering, why would you need that?

The idea is you access the entries using the method `getEntry()`.

**Comment by gorets** [comment]
I'm doing an online PO-Editor.
In web page loaded list all items in file.
By Ajax request loaded current item in form edit.
I want to send only unique key in ajax-request, and not 2 parameters (msgid & msgctxt) to get the current item.

Or you can modify the function:

```
public function removeEntry($msgid, $msgctxt = null, $key = null)
{
	if(!$key){
		$key = $this->getEntryHash($msgid, $msgctxt);
	}
	if (isset($this->entries[$key])) {
		unset($this->entries[$key]);
	}
}
```

**Comment by raulferras** [comment]
I understand your use case.  
One question, how are you getting this key currently?

**Comment by gorets** [comment]
Sample use:
```
<?php
$entries = $catalog->getEntries();
foreach ($entries as $key => $entry) {
?>
<tr data-key="<?= HTML::encode($key) ?>">
    <td>
        <?= HTML::encode($entry->getMsgId()) ?>
    </td>
</tr>
<?}?>

<script>
    $("tr").on('click', function(){
        var $row = $(this);
        var key = $row.data('key');
        if (key) {
            $.pjax.reload('#form_wrap', {
                'url': '/index.php?r=po-edit/get-item',
                'data': {key: key},
            });
        }
    })
</script>
```

**Comment by raulferras** [comment]
Addressed in #72
